operator/controllers/btpoperator_controller.go

```diff
@@ -502,19 +504,31 @@ func (r *BtpOperatorReconciler) handleDeprovisioning(ctx context.Context) error
 			logger.Info("hard delete success")
 			if err := r.cleanUpAllBtpOperatorResources(ctx, namespaces); err != nil {
 				logger.Error(err, "failed to remove related installed resources")
-				return err
+				return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, ResourceRemovalFailed, "Unable to remove installed resources")
 			}
 		} else {
-			if err := r.handleSoftDelete(ctx, namespaces); err != nil {
+			err := r.UpdateBtpOperatorStatus(ctx, cr, types.StateDeleting, SoftDeleting, "Being soft deleted")
+			if err != nil {
+				logger.Error(err, "failed to update status")
 				return err
 			}
+			if err := r.handleSoftDelete(ctx, namespaces); err != nil {
+				logger.Error(err, "failed to hard delete")
+				return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, HardDeleteFailed, "Hard deleting failed")
```

Comment:
if we get error from `handleSoftDelete` then it's soft delete error. Hard delete fail is an intermediate state, if BtpOperatorReconciler is processing soft delete, then we know hard delete failed

Reply:
Corrected in the next commits.